compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -443,9 +447,10 @@ public BType resolvePatternTypeFromMatchExpr(BLangListMatchPattern listMatchPatt
         if (listMatchPattern.matchExpr == null) {
             return listMatchPatternType;
         }
-        BType matchExprType = listMatchPattern.matchExpr.getBType();
+        BLangExpression matchExpr = listMatchPattern.matchExpr;
+        BType matchExprType = matchExpr.getBType();
         BType intersectionType = getTypeIntersection(
-                IntersectionContext.compilerInternalIntersectionContext(),
+                IntersectionContext.compilerInternalIntersectionContext(listMatchPattern.pos, matchExpr.pos),
```

Comment:
In this scenario, isn't the start pos after the end pos?

Reply:
I think what we have done is OK for the type test, but for match it may not be ideal. Because there will be overlaps if we use matchExpr pos and pattern pos. 

The other option I guess to introduce another pos, maybe `intersectionPos` and set a pos we decide to be the intersection's pos to that directly.

So in the case of type tests, it'll be the pos of the type test expression itself, and for match the pos of the pattern?

Reply:
Introduced `intersectionPos`